src/metabase/api/card.clj

```diff
@@ -669,5 +669,4 @@
   (api/check-embedding-enabled)
   (db/select [Card :name :id], :enable_embedding true, :archived false))
 
-(api/define-routes
-  (middleware/streaming-json-response (route-fn-name 'POST "/:card-id/query")))
+(api/define-routes)
```

Comment:
@metabase/core-developers this middleware was (incorrectly) applied here before. I have added the query cancellation logic to queries made via the dataset route. Should I be adding it to `/card/:card-id/query` as well?